### ISSUE TITLE
Refund Transaction creates transaction with all zero amounts 

### ISSUE THREAD

**Issue by rwhirn** [avadev/AvaTax-REST-V2-PHP-SDK#145]
The Refund Transaction method is just not that complicated, you give it the old transaction code, the doc type, SalesInvoice , then in the body a new transaction code, the date and the refund type, partial, full or tax only. There's not a whole lot I can be doing wrong here, but every time I try I get a perfect copy of the old transaction (as expected, all the lines are there and everything) but all the amounts are 0 (and all the quantities on the lines are reset to 1). What is going wrong?

**Comment by rwhirn** [comment]
The call to the API in the SDK looks like this:
`Array
(
    [query] => Array
        (
            [$include] => 
            [documentType] => 1
            [useTaxDateOverride] => true
        )

    [body] => {"refundTransactionCode":"1320108-REFUND","refundDate":"2021-03-27","refundType":2,"refundPercentage":null,"refundLines":null,"referenceCode":"Tax refunded"}
    [auth] => Array
        (
            [0] => 
            [1] => 
        )

    [headers] => Array
        (
            [Accept] => application/json
            [X-Avalara-Client] => FYPapp; 1.0; PhpRestClient; 20.1.0; localhost
        )

)`

But the examples in the Postman collection looks like this:
`https://rest.avatax.com/api/v2/companies/DEFAULT/transactions/1319742/refund
{"refundTransactionCode":"1319742-REFUND","refundDate":"2021-08-16","refundType":"Full","referenceCode":"Refund for a committed transaction"}
`

The Postman example works, but notice `"refundType":"Full"`

Why does the SDK send an integer?

**Comment by rwhirn** [comment]
BTW, what values can be entered for "refundType" is not documented anywhere in the documentation, neither [https://developer.avalara.com/api-reference/avatax/rest/v2/methods/Transactions/RefundTransaction/](https://developer.avalara.com/api-reference/avatax/rest/v2/methods/Transactions/RefundTransaction/)
nor Swagger [https://sandbox-rest.avatax.com/swagger/ui/index.html#/Transactions/RefundTransaction](https://sandbox-rest.avatax.com/swagger/ui/index.html#/Transactions/RefundTransaction)
The only place I found it mentioned was in [https://developer.avalara.com/static/pdf/AvaTaxAPIDeveloperGuide2018.pdf](https://developer.avalara.com/static/pdf/AvaTaxAPIDeveloperGuide2018.pdf)

> You can choose to use the RefundTransaction API in a few different ways:

> •• You can refund the entire transaction by setting “refundType”: “Full”.

> •• You can refund specific lines from a transaction by specifying “refundType”: “Partial”, and setting a list of lines to refund using the refundLines parameter to the list of line numbers of the lines that are being refunded.

> •• If your customer originally paid sales tax on the transaction, and they later provided a resale exemption certificate, you can call set the “refundType”: “TaxOnly”.

> •• If you wish to give a customer a percentage refund, perhaps for a discount, you can specify “refundType”: “Percentage”.

**Comment by svc-developer** [comment]
After investigation we found that ENUMS/Constants work fine in PHP SDK. If we pass the ENUMS as String Constant it works and also if we pass ENUM Value as String that works as well.

So in the above example, if we pass "refundType":"2" or "refundType":"TAXONLY" then it will work as expected. 

However if we pass ENUM value as INTEGER (like "refundType":2), it doesn’t work. We agree that some of the documentation needs to be more clear about this. 

Closing the ticket for now.
